menuconfig.py

```diff
@@ -299,6 +299,173 @@
+    """
+    # The formula for converting a single RGB value to the 6cube index
+    # 48 is the middle ground between 0 and 95
+    rgb_to_c6 = lambda x: 0 if x < 48 else max(1, int(round((x - 55) / 40)))
+
+    return rgb_to_c6(r), rgb_to_c6(g), rgb_to_c6(b)
+
+def _rgb_from_6cube(r6, g6, b6):
+    """
+    Take a 666 xterm color cube index and convert it to a
+    3 tuple representing the RGB color
+    """
+    # The formula from converting the xterm 6cube index to a RGB value
+    rgb_from_c6 = lambda x: 0 if x == 0 else 55 + x*40
+
+    return rgb_from_c6(r6), rgb_from_c6(g6), rgb_from_c6(b6)
+
+def _rgb_to_gray_idx(r, g, b):
+    """
+    Convert an 888 RGB color to the index of an xterm 256-color grayscale color
+    with approx. the same perceived brightness. This "grayscale candidate" can
+    be compared against the best "color candidate" to find a good xterm color
+    to represent the color.
+    """
+    # Calculate the luminance (gray intensity) of a color from its R, G, B components
+    # Source: https://stackoverflow.com/questions/596216/formula-to-determine-brightness-of-rgb-color
+    luma = 0.299 * r + 0.587 * g + 0.114 * b
+
+    # Closests index in the grayscale palette, which starts at RGB 0x080808, with
+    # stepping 0x0A0A0A
+    idx = int(round((luma - 0x08) / 0x0A))
+
+    # Clamp the index to 0-23, corresponding to 232-255
+    return max(0, min(idx, 23))
+
+def _rgb_from_gray_idx(c):
+    """
+    Convert a grayscale index to its closet single RGB component
+    """
+    g = c * 10 + 8
+    return (g, g, g)
+
+# Obscure Python: rgb2index is initialized from a reference to a global {}
+# Modification to this dictionary are retained between calls rgb_alloc(), thus
+# making rgb2index a static variable
+def _rgb_alloc(rgb, rgb2index={}):
+    """
+    Initialize a new entry in the xterm palette to the given RGB color, returning its
+    index. If the color has already been initialized, the index of the existing entry is
+    returned.
+
+    Ncurses doesn't allow you to define new colors -- you are allowed only
+    to overwrite existing ones.
+
+    The colors from 0-15 are user-defined and there's no way to query the
+    RGB values so we better leave them untouched.
+
+    The RGB values of colors from 16-255 can be easily calculated. However,
+    colors from 232-255 use a different formula (grayscale) so stick with
+    colors from 16-231. 200+ colors should be plentiful.
+    """
+    if rgb in rgb2index:
+        return rgb2index[rgb]
+
+    # The first 16 colors are user defined and we should not change their
+    # values
+    cn = len(rgb2index) + 16
+    if cn >= curses.COLORS:
+        _warn("Unable to allocate new RGB color.", rgb)
+        return 0
+
+    # Map each RGB component from the range 0-255 to the range 0-1000, which
+    # is what curses uses
+    curses.init_color(cn, *(int(round(x * 1000 / 255)) for x in rgb))
+    rgb2index[rgb] = cn
+
+    return cn
+
+def _color_get(num):
+    """
```

Comment:
For internal functions, the convention is to use "doc comments" instead of docstrings. See e.g. `_is_num()`.

Ditto for other functions.